client/client_api.c

```diff
@@ -0,0 +1,128 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <inttypes.h>
+#include <string.h>
+#include "client_api.h"
+/************************************************
+ *                     GW KV
+ *  https://github.com/gwAdvNet2015/gw-kv-store
+ *
+ * Copyright 2015 Lucas Chaufournier, Joel Klein,
+ * and Katie Stasaski
+ *
+ * This program is licensed under the MIT license.
+ *
+ * client_api.c- creates client api and modularizes
+ *************************************************/
+int
+set_mem(int sockfd, struct operation * msg)
+{
+    char * temp;
+    int num_bytes, rc;
+    int status = 0;
+
+    gwkv_marshal_client(msg, &temp);
+
+    rc = send(sockfd,temp,strlen(temp), 0);
+    if(rc < 0) {
+        status = -1;
+    }
+
+    return status;
+}
+
+int
+get_mem(int sockfd, struct operation * msg)
+{
+    char * temp;
+    int num_bytes, rc;
+    int status = 0;
+
+    gwkv_marshal_client(msg, &temp);
+
+    rc = send(sockfd,temp,strlen(temp), 0);
+    if(rc < 0) {
+        status = -1;
+        return status;
+    }
+
+    status = read_get_msg(sockfd, msg);
+
+    return status;
+}
+
+int
+read_get_msg(int sockfd, struct operation * msg)

```

Comment:
Please add the __ prefix before this function to state it's an internal function.